graph/memstore/quadstore_test.go

```diff
@@ -251,3 +256,42 @@ func TestTransaction(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, st, st2, "Appended a new quad in a failed transaction")
 }
+
+// test multi thread insert and query
+func TestMultiThreadQuery(t *testing.T) {
+	qs, _, _ := makeTestStore(simpleGraph)
+
+	// we make 50 insert, 50 query
+	funcs := make([]async.LambdaMethod, 100)
+	for i := 0; i < 100; i++ {
+		if i%2 == 0 {
+			index := i
+			funcs[i] = func() (interface{}, error) {
+				id, flag := qs.AddQuad(quad.Make(
+					fmt.Sprintf("E_%d", index), "follows", "G", nil),
+				)
+				if !flag {
+					return nil, fmt.Errorf("quard exist:%d", id)
+				}
+				return id, nil				
+			}
+		} else {
+			funcs[i] = func() (interface{}, error) {
+				ctx := context.Background()
+				followers, err := path.StartPath(qs, quad.Raw("G")).In("follows").Iterate(ctx).AllValues(qs)
+				if err != nil {
+					return nil, err
+				}
+				return followers, nil
+			}
+		}
+	}
+
+	results := async.All(funcs, 1*time.Second)
```

Comment:
Would be also nice to test results for consistency somehow, but that might not be trivial. Could be done later.